treeherder/auth/backends.py

```diff
@@ -1,83 +1,145 @@
+        cache.set(cache_key, cached_jwks)
+
+        return cached_jwks
+
+    def _validate_token(self, request):
+        token = self._get_token_auth_header(request)
+        # JWT Validator
+        # Per https://auth0.com/docs/quickstart/backend/python/01-authorization#create-the-jwt-validation-decorator
+        jwks_json = self._get_jwks_json()
+        jwks = json.loads(jwks_json)
 
-        tc_auth = Auth()
-        # https://docs.taskcluster.net/reference/platform/taskcluster-auth/references/api#authenticateHawk
-        # https://github.com/taskcluster/taskcluster-client.py#authenticate-hawk-request
-        result = tc_auth.authenticateHawk({
-            "authorization": auth_header,
-            "host": host,
-            "port": port,
-            "resource": reverse("auth-login"),
-            "method": "get",
-        })
+        unverified_header = jwt.get_unverified_header(token)
 
-        if result["status"] != "auth-success":
-            logger.warning("Error logging in: {}".format(result["message"]))
-            raise TaskclusterAuthException(result["message"])
+        rsa_key = {}
+        for key in jwks["keys"]:
+            if key["kid"] == unverified_header["kid"]:
+                rsa_key = {
+                    "kty": key["kty"],
+                    "kid": key["kid"],
+                    "use": key["use"],
+                    "n": key["n"],
+                    "e": key["e"]
+                }
 
-        client_id = result["clientId"]
-        email = self._extract_email_from_clientid(client_id)
+        if not rsa_key:
+            raise AuthError({"code": "rsa_key",
+                            "description": "rsa_key is empty"}, 401)
+
+        try:
+            jwt.decode(
+                token,
+                rsa_key,
+                algorithms=['RS256'],
+                audience=AUTH0_AUDIENCE,
+                issuer="https://"+AUTH0_DOMAIN+"/"
+            )
+        except jwt.ExpiredSignatureError:
+            raise AuthError({"code": "token_expired",
+                            "description": "token is expired"}, 401)
+        except jwt.JWTClaimsError:
+            raise AuthError({"code": "invalid_claims",
+                            "description":
+                                "incorrect claims,"
+                                "please check the audience and issuer"}, 401)
+        except Exception:
+            raise AuthError({"code": "invalid_header",
+                            "description":
+                                "Unable to parse authentication"
+                                " token."}, 400)
+
+    def _get_user_info(self, request):
+        user_info_url = 'https://' + AUTH0_DOMAIN + '/userinfo'
+
+        return json.loads(
+            urlopen(Request(user_info_url, headers={'Authorization': request.META.get('HTTP_AUTHORIZATION', None)})).read())
+
+    def authenticate(self, request):
+        self._validate_token(request)
```

Comment:
So I think this is not *strictly* necessary for an access_token, since the next step is to pass it to `/userinfo`, which will also validate it.  OIDC also doesn't guarantee that access_token will be a JWT, so it feels a little funny to be making this check.